moped-editor/src/views/projects/projectsListView/ProjectsListViewTable.js

```diff
@@ -538,6 +538,13 @@ const ProjectsListViewTable = ({ query, searchTerm }) => {
     query.clearOrderBy();
     const columnName = columns[columnId]?.field;
 
+    // Resets pagination to 0 when user clicks a header to display most relevant results
+    setPagination({
+      limit: query.limit,
+      offset: query.offset,
+      page: 0,
+    });
```

Comment:
i would think we need to also touch `query.offset` to reset that back to `0` as well. but that's not the case. i spent a little bit of time trying to understand why this works, and i couldn't grok it. all that to say - 👍 thanks for figuring out how to patch this!

Reply:
this drove me nuts a little, so I did some digging. I think what is happening is the setPagination state update that happens when someone clicks next on the page at the bottom does update the query.offset in GQLAbstract to reflect the correct offset, which is then put into the query that is generated by GQLAbstract. 

Then the whole page refreshes which has the side effect of resetting the state to have the offset from the config, i.e. 0. But doesn't update the query itself. 

Now how come it works on page 3? I ask myself the same question and I think the pagination state is *actually* being kept track of in the TablePagination component thats in the GridTablePagination component. and now i will stop digging around in this.

Reply:
thanks y'all for digging into this rabbit hole with me! :p it's been driving me crazy too. but i'm glad this fix is ready to ship. maybe we should revisit this as tech debt at some point?

Reply:
+1 thanks for that research @chiaberry!

yes, we'll definitely revisit. i haven't been sure how to write up issues related to this, but i'll capture it somewhere. i'm particularly excited about [this plugin](https://github.com/hasura/ra-data-hasura) that @mddilley found that looks to me like a replacement for the graphql abstract.